### ISSUE TITLE
Support `when.repo` filter

### ISSUE THREAD

**Issue by anbraten** [woodpecker-ci/woodpecker#687]
It would be nice to have an option to filter if the CI should be executed for a specific repo. This would allow to not automatically run CI on forks. 

How it could look like.

```yml
when:
  repo: my/main-repo
  branch: main
```

To keep things simple I would personally prefer to implement this by #312

**Comment by vanous** [comment]
Again, thank you for all the work on this infrastructural piece, which allows us to use this for our work on another pieces of the FLOSS ecosystem!

Please correct me if i am wrong, but the docs mention that this is already supported:

![image](https://user-images.githubusercontent.com/3680926/149614789-e416c80b-6eee-4abb-8e86-557343f4f5cd.png)
https://woodpecker-ci.org/docs/usage/conditional-execution#repo

I am trying to apply a `repo` condition to the whole pipeline, which also should be supported:

![image](https://user-images.githubusercontent.com/3680926/149614888-186afb51-dd3f-4199-b312-bd0f50b8180b.png)
https://woodpecker-ci.org/docs/usage/pipeline-syntax#when

But when i move `when` to the root level, it is still executed:

```yaml
can_master_build
pipeline:
  build-apk:
    image: androidsdk/android-30
    commands:
      - pwd #bump
      - ./gradlew assembleDebug --stacktrace
when:
  repo: notme/notmyrepo
```
I can do this on a step level, but then the CI actually runs, pulls the repo, skips the conditional step and finished with a success, which can be misleading.

So maybe the docs will need to be (temporarily) updated?

**Comment by vanous** [comment]
OK, i think this PR https://github.com/woodpecker-ci/woodpecker/pull/642 did not consider pipeline config, they used it most likely in step config. cc: @nupplaphil

**Comment by nupplaphil** [comment]
From my point of view, there are several `when` cases, which aren't working at `root` level. That's why I moved all root conditions into the steps and repeated them for every step.

So yes, the `repo` doesn't work at root level. I hope the the refactoring of the `when` block as you mentioned will fix the overall behavior.

If you need concrete conditions, which weren't working for me, I could reproduce them and create an issue for it, but as I said I found a workaround and found the task for "overall" when refactoring so I didn't mentioned it yet

**Comment by vanous** [comment]
Thank you. 

> So yes, the `repo` doesn't work at root level. I hope the the refactoring of the `when` block as you mentioned will fix the overall behavior.

The docs are currently unfortunately misleading.

**Comment by anbraten** [comment]
Missed that it already exists for steps. So we are "only" missing this option for complete pipelines which will be solved by #283
